pkg/crypto/interface.go

```diff
@@ -12,5 +12,14 @@ import (
 // the addresses of the gateways the application is delegated to, and converting
 // them into their corresponding public key points on the secp256k1 curve.
 type RingCache interface {
+	// Start starts the ring cache, it takes a cancellable context and, in a
+	// separate goroutine, listens for on-chain delegation events and invalidates
+	// the cache if the delegatee change's address is stored in the cache.
```

Comment:
Wdyt about putting quotes around "delegatee change" so as not to misread it as "delegatee changes" (i.e. changes as a verb instead of part of the noun which is the object in question):

```suggestion
	// the cache if the "delegatee change"'s address is stored in the cache.
```

Reply:
After thinking about it a bit more and getting a bit deeper into this PR, I find myself wanting to suggest renaming the concept of the  "delegatee change" event to a "redelegate" event. I think the fact that it's a one word noun simplifies a lot.

@h5law, I'm noticing references to both "delegatee change" and "delegation", are these interchangable or do they have distinct meanings?